src/main/java/ua/artcode/billapp/security/JwtAuthTokenFilter.java

```diff
@@ -0,0 +1,41 @@
+package ua.artcode.billapp.security;
+
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.AuthenticationException;
+import org.springframework.security.web.authentication.AbstractAuthenticationProcessingFilter;
+import ua.artcode.billapp.model.JwtAuthToken;
+
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+public class JwtAuthTokenFilter extends AbstractAuthenticationProcessingFilter {
+
+
+    public JwtAuthTokenFilter() {
+        super("/rest/**");
+    }
+
+    @Override
+    public Authentication attemptAuthentication(HttpServletRequest request, HttpServletResponse response) throws AuthenticationException {
+
+        String header = request.getHeader("Authorisation");
+
+        if (header == null || !header.startsWith("Token ")){
+            throw new RuntimeException("JWT Token is missing");
```

Comment:
Did you saw what Codacy say? "Avoid throwing raw exception types."
I didn't like this system (Codacy), but now it says right thing: AVOID RUNTIME EXCEPTIONS.

It is time to use violence to everyone, who use RuntimeException to avoid "throws" and "try/catch" block.